test/extended/testdata/build-extended/bc-scripts-in-the-image.json

```diff
@@ -0,0 +1,34 @@
+{
+  "kind": "BuildConfig",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "java-extended-build-from-image"
+  },
+  "spec": {
+    "source": {
+      "git": {
+        "uri": "https://github.com/php-coder/java-maven-hello-world.git",
+        "ref": "s2-assemble-only"
+      }
+    },
+    "strategy": {
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "wildfly:10.0",
+          "namespace": "openshift"
+        },
+        "runtimeImage": {
+          "kind": "DockerImage",
+          "name": "phpcoder/hello-world-runtime:0.2"
```

Comment:
Still not seeing why you can't use that image here too.

put all the scripts in the image.  your tests are already confirming that the "right" script got used by checking output, so it's ok to use an image that includes an assemble-runtime script, because your test should be confirming it didn't get used if the user supplied a script.